layers/vk_layer_data.h

```diff
@@ -198,6 +198,18 @@ class small_vector {
         return true;
     }
 
+    bool operator!=(const small_vector &rhs) const {
+        if (size_ != rhs.size_) return true;
+        auto value = begin();
+        for (const auto &rh_value : rhs) {
+            if (!(*value == rh_value)) {
+                return true;
+            }
+            ++value;
+        }
+        return false;
+    }
```

Comment:
Couldn't we leverage the existing `==` operator?

```suggestion
    bool operator!=(const small_vector &rhs) const { return !(*this == rhs); }
```